### PR TITLE
Hotfix: Fix release string comparison for RHEL8/CentOS8 modular packages

### PR DESCRIPTION
**Pull request by hustodemon**
The current comparing algorithm evaluates the 'release' part of some rhel8 modular rpms as equal, which breaks other parts of Uyuni (reposyncing a rhel/centos8 channels leads to unique constraint violation on `rhnChannelNewestPackage`).

This PR is not the ultimate fix to the problem, it only restores the old behavior (before https://github.com/uyuni-project/uyuni/pull/2615), where the constraint violation didn't happen (packages were ordered (although very likely incorrectly)).

Currently, there is an effort to improve the version comparison (split the rpm and deb logic). When it's done, this PR should be reverted!

## GUI diff

No difference.

- [x] **DONE**

## Documentation
fix

## Test coverage
- Unit tests were added

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mbologna** [comment]
Should migration file be present?

File: java/spacewalk-java.changes [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Hotfix the modular RPMs release comparison
```
Comment (mbologna) [inline java/spacewalk-java.changes:1]: a bsc# is not present, is this intended?

Comment (hustodemon) [reply java/spacewalk-java.changes:1]: there is no bsc

**Review by mbologna — APPROVED** [review]

**Comment by mbologna** [comment]
LGTM. There are some failing tests but they are related to the checks API

**Review by mcalmer — CHANGES_REQUESTED** [review]
DB schema migration still missing

**Comment by hustodemon** [comment]
Added migration scripts & rebased.

**Comment by hustodemon** [comment]
(I also verified that syncing centos8 appstream channel does not fail anymore.)

**Review by mcalmer — APPROVED** [review]

**Pull request merged**